### PR TITLE
fix: position correct amount & vault manager debt amount & usePosition hook

### PR DESCRIPTION
**Pull request by yichiehliu**
# Task:

## Description

Fix:
1. Showed correct values for short squeeth, minted squeeth, LPed squeeth
2. Moved custom debt logic from vault manager, positions, and short component to usePositions
3. Changed to swap subscription in usePosition to update data fast enough for user, and dont need to refresh to see their positions
4. Fixed BN Object in dep array issue
5. Changed oSqthBal to longSqthBal, which is user osquth balance subtracted by minted oSQTH amount, for getting correct long balance in long trade ticket

Feat:
1. Added LPed amount on vault manager page
image
Fixes # (issue) #62 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested
Locally

Please describe how to test to verify the changes. Provide instructions so we can reproduce.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Added video recordings if it is a UI change

### CODE REVIEW

File: packages/frontend/src/hooks/usePositions.ts [hunk 1]
```diff
@@ -41,7 +42,28 @@ export const usePositions = () => {
...
+    subscribeToNewPositions()
+  }, [])
+
+  const subscribeToNewPositions = useCallback(() => {
+    subscribeToMore({
+      document: SWAPS_SUBSCRIPTION,
+      variables: {
+        poolAddress: squeethPool?.toLowerCase(),
+        origin: address || '',
+        recipients: [shortHelper, address || '', swapRouter],
```
Comment (alexisgauba) [inline packages/frontend/src/hooks/usePositions.ts:55]: this adjust the query to get swaps from short helper and the Uniswap swap router for individual sells after mints?

File: packages/frontend/src/hooks/usePositions.ts [hunk 2]
```diff
@@ -112,12 +134,33 @@ export const usePositions = () => {
...
+  const lpDebt = useMemo(() => {
+    return depositedSqueeth.minus(withdrawnSqueeth).isGreaterThan(0)
+      ? depositedSqueeth.minus(withdrawnSqueeth)
+      : new BigNumber(0)
+  }, [positionType, depositedSqueeth.toString(), withdrawnSqueeth.toString()])
+
   const { finalSqueeth, finalWeth } = useMemo(() => {
-    const finalSqueeth = squeethAmount.minus(depositedSqueeth).plus(withdrawnSqueeth)
+    // dont include LPed amount will be the correct short amount
+    const finalSqueeth = squeethAmount
```
Comment (alexisgauba) [inline packages/frontend/src/hooks/usePositions.ts:159]: wdym correct short amount? If you are just looking for how much someone is short that should not include both minted and LPed

**Review by alexisgauba — COMMENTED** [review]
Overall looking really good! 

Just a few things 
- Would seperate out LP 
- Looks like PnL is showing NaN now 
Screen Shot 2022-01-19 at 5 22 29 PM

File: packages/frontend/src/hooks/usePositions.ts (hunk 1 above)
Comment (yichiehliu) [reply packages/frontend/src/hooks/usePositions.ts:55]: Oh, this adjust is for subscription, I didn't change the swaps pulling logics

File: packages/frontend/src/hooks/usePositions.ts (hunk 2 above)
Comment (yichiehliu) [reply packages/frontend/src/hooks/usePositions.ts:159]: But the squeethAmount is already the short amount that someone has, the LPed and minted amount was not accounted as squeethAmount in the beginning so we dont need to subtract LPed and minted amount here

**Comment by yichiehliu** [comment]
Right now it should be able to separate out LPed amount from minted amount, it doesn't work for you?
image

**Comment by alexisgauba** [comment]
> Right now it should be able to separate out LPed amount from minted amount, does it work for you? image

On the vault manager it groups minted and LP together which isn't accurate, since they are both very distinct positions

File: packages/frontend/src/hooks/usePositions.ts (hunk 1 above)
Comment (alexisgauba) [reply packages/frontend/src/hooks/usePositions.ts:55]: Ok don't we need to adjust the swap query to make sure we get all Uniswap swaps not just ones from our short helper?

File: packages/frontend/src/hooks/usePositions.ts (hunk 2 above)
Comment (alexisgauba) [reply packages/frontend/src/hooks/usePositions.ts:159]: Ok, comment was confusing because it only called out LP not minting as well

Comment (yichiehliu) [reply packages/frontend/src/hooks/usePositions.ts:159]: lol gotcha, will edit this

**Comment by EstherSeyi** [comment]
Looks good to me asides from the things that have already been pointed out.

File: packages/frontend/src/hooks/usePositions.ts (hunk 1 above)
Comment (yichiehliu) [reply packages/frontend/src/hooks/usePositions.ts:55]: but do we really miss some txs here?

**Comment by alexisgauba** [comment]
Screen Shot 2022-01-20 at 10 05 46 AM
we'd need to add back showing LP squeeth on the vault manager as well!

File: packages/frontend/src/hooks/usePositions.ts (hunk 1 above)
Comment (alexisgauba) [reply packages/frontend/src/hooks/usePositions.ts:55]: yeah if a user buys back on Uniswap directly it won't capture that right? Or if they just mint and then go sell

Comment (yichiehliu) [reply packages/frontend/src/hooks/usePositions.ts:55]: Ah kk, maybe I  fix this issue in another PR?

**Review by alexisgauba — APPROVED** [review]
lgtm!

**Review by KMKoushik — APPROVED** [review]

**Pull request merged**